packages/eventual-send/src/handled-promise.js

```diff
@@ -548,8 +553,9 @@ export const makeHandledPromise = () => {
 
     // Harden the fulfillment and rejection, as well as a workaround for
     // Node.js: silence "Unhandled Rejection" by default when using the static
-    // methods.
-    returnedP.then(harden, harden);
+    // methods. Use a void returning version of harden to prevent transforming
+    // a rejected promise used as reason into a new unhandled rejection.
+    returnedP.then(hardenVoid, hardenVoid);
```

Comment:
@michaelfig if the intent is to NOT silence unhandled rejections, then the body of the HandledPromise constructor above should call harden before passing to `resolve` or `reject`.

Reply:
I see.  Maybe a separation of orthogonal concerns would be clearest?  `harden(reason)` and `harden(fulfillment)`, as well as just a `returnedP.catch(() => {})`?

WDYT?

Reply:
Oh wait, I want to harden the settled value.  So yes, your code is probably best.

Reply:
> if the intent is to NOT silence unhandled rejections, then the body of the HandledPromise constructor above should call `harden` before passing the value to `resolve` or `reject`.

I have a counterproposal along those lines in #1178.  Would appreciate your eyes on it.